blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineRunImpl.java

```diff
@@ -155,13 +154,9 @@ public void stop() {
 
     @Exported(name = "commitId")
     public String getCommitId() {
-        BuildData data = run.getAction(BuildData.class);
-
+        SCMRevisionAction data = run.getAction(SCMRevisionAction.class);
         if (data != null){
-            Revision revision = data.getLastBuiltRevision();
-            if(revision != null){
-                return revision.getSha1String();
-            }
+            return data.getRevision().toString();
```

Comment:
Of course the details of what `SCMRevision.toString` will return is up to the SCM. For Git this will generally be a commit hash, though it will be something else in the case of a GitHub PR merge.